[openctm] Cannot link with both 7zip and openctm

### Is your feature request related to a problem? Please describe.

Openctm vendors parts of 7zip (LZMA-SDK to be precise). To avoid ODR violations it changes function names. Unfortunately it does not change all names and we end up with conflicts like these:

```
/usr/bin/ld.gold: error: /g/mr/vcpkg_installed/x64-linux-cxx20-rel/lib/lib7zip.a(LzmaEnc.c.o): multiple definition of 'LzmaEncode'
/usr/bin/ld.gold: /g/mr/vcpkg_installed/x64-linux-cxx20-rel/lib/libopenctm.a(LzmaEnc.c.o): previous definition here
```

For projects that use both `openctm` and `7zip`.

### Proposed solution

Patch openctm (which is unmaintained btw., so no use in creating upstream ticket) such that it uses the 7zip library provided by vcpkg. The patch is rather simple (see below)

### Describe alternatives you've considered

_No response_

### Additional context

To patch openctm:

Remove `LZMA_PREFIX_CTM` and add find_package etc. for 7zip in CMakeLists.txt

Adjust source code to the more recent version of 7zip:

```patch
diff --git a/lib/stream.c b/lib/stream.c
index 53b1b72..bb47c3b 100644
--- a/lib/stream.c
+++ b/lib/stream.c
@@ -310,8 +310,7 @@ int _ctmStreamWritePackedInts(_CTMcontext * self, CTMint * aData,
                          outProps,
                          &outPropsSize,
                          self->mCompressionLevel, // Level (0-9)
-                         0, -1, -1, -1, -1, -1,   // Default values (set by level)
-                         lzmaAlgo                 // Algorithm (0 = fast, 1 = normal)
+                         0, -1, -1, -1, -1, -1   // Default values (set by level)
                         );
 
   // Free temporary array
@@ -477,8 +476,7 @@ int _ctmStreamWritePackedFloats(_CTMcontext * self, CTMfloat * aData,
                          outProps,
                          &outPropsSize,
                          self->mCompressionLevel, // Level (0-9)
-                         0, -1, -1, -1, -1, -1,   // Default values (set by level)
-                         lzmaAlgo                 // Algorithm (0 = fast, 1 = normal)
+                         0, -1, -1, -1, -1, -1   // Default values (set by level)
                         );
 
   // Free temporary array

```

--- a/ports/openctm/CMakeLists.txt
+++ b/ports/openctm/CMakeLists.txt
@@ -4,6 +4,8 @@ project(
   VERSION 1.0.3
   LANGUAGES C)
 
+find_package(7zip CONFIG REQUIRED)
+
 set(PUBLIC_HEADERS lib/openctm.h lib/openctmpp.h)
 
 add_library(
@@ -12,19 +14,14 @@ add_library(
   lib/compressMG2.c
   lib/compressRAW.c
   lib/openctm.c
-  lib/stream.c
-  lib/liblzma/Alloc.c
-  lib/liblzma/LzFind.c
-  lib/liblzma/LzmaDec.c
-  lib/liblzma/LzmaEnc.c
-  lib/liblzma/LzmaLib.c)
+  lib/stream.c)
 
 target_include_directories(
   openctm
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/lib>
-         $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/lib/liblzma>
          $<INSTALL_INTERFACE:include>)
-target_compile_definitions(openctm PRIVATE OPENCTM_BUILD LZMA_PREFIX_CTM)
+target_link_libraries(openctm PRIVATE 7zip::7zip)
+target_compile_definitions(openctm PRIVATE OPENCTM_BUILD)
 if(NOT BUILD_SHARED_LIBS)
   target_compile_definitions(openctm PUBLIC OPENCTM_STATIC)
 endif()

--- a/ports/openctm/portfile.cmake
+++ b/ports/openctm/portfile.cmake
@@ -4,6 +4,8 @@ vcpkg_from_sourceforge(
     REF OpenCTM-1.0.3
     SHA512 fdfa08d19ecbfea99ba01aa2032e941ed6313394a96bd69f8984c2d2d079d836c616471d2bdf6f40175e75659f3ad0ba41502bc3d8224091472f40893ea8746e
     FILENAME "OpenCTM-1.0.3-src.tar.bz2"
+    PATCHES
+        use-7zip.patch
 )
 
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
@@ -18,4 +20,4 @@ vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-openctm)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.txt")

--- a/ports/openctm/use-7zip.patch
+++ b/ports/openctm/use-7zip.patch
@@ -1,0 +1,24 @@
+diff --git a/lib/stream.c b/lib/stream.c
+index 53b1b72..bb47c3b 100644
+--- a/lib/stream.c
++++ b/lib/stream.c
+@@ -310,8 +310,7 @@ int _ctmStreamWritePackedInts(_CTMcontext * self, CTMint * aData,
+                          outProps,
+                          &outPropsSize,
+                          self->mCompressionLevel, // Level (0-9)
+-                         0, -1, -1, -1, -1, -1,   // Default values (set by level)
+-                         lzmaAlgo                 // Algorithm (0 = fast, 1 = normal)
++                         0, -1, -1, -1, -1, -1   // Default values (set by level)
+                         );
+ 
+   // Free temporary array
+@@ -477,8 +476,7 @@ int _ctmStreamWritePackedFloats(_CTMcontext * self, CTMfloat * aData,
+                          outProps,
+                          &outPropsSize,
+                          self->mCompressionLevel, // Level (0-9)
+-                         0, -1, -1, -1, -1, -1,   // Default values (set by level)
+-                         lzmaAlgo                 // Algorithm (0 = fast, 1 = normal)
++                         0, -1, -1, -1, -1, -1   // Default values (set by level)
+                         );
+ 
+   // Free temporary array

--- a/ports/openctm/vcpkg.json
+++ b/ports/openctm/vcpkg.json
@@ -1,10 +1,12 @@
 {
   "name": "openctm",
   "version": "1.0.3",
+  "port-version": 1,
   "description": "OpenCTM — the Open Compressed Triangle Mesh file format — is a file format, a software library and a tool set for compression of 3D triangle meshes.",
   "homepage": "http://openctm.sourceforge.net/",
   "license": "Zlib",
   "dependencies": [
+    "7zip",
     {
       "name": "vcpkg-cmake",
       "host": true

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6714,7 +6714,7 @@
     },
     "openctm": {
       "baseline": "1.0.3",
-      "port-version": 0
+      "port-version": 1
     },
     "opencv": {
       "baseline": "4.10.0",

--- a/versions/o-/openctm.json
+++ b/versions/o-/openctm.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "92ac7a53f30a5fadc0cf5186f2d256c87d494471",
+      "version": "1.0.3",
+      "port-version": 1
+    },
+    {
       "git-tree": "ea30144a8ad57cb95b0603de14e4dc8e9f7b2ea2",
       "version": "1.0.3",
       "port-version": 0

